pkg/cover/html.go

```diff
@@ -212,8 +219,29 @@ func (rg *ReportGenerator) DoRawCoverFiles(w io.Writer, params CoverHandlerParam
 	return nil
 }
 
+type coverFileDetails struct {
+}
+
+type coverDatails struct {
+	Version int `json:"version"`
+	Files   []coverFileDetails
+}
+
 func (rg *ReportGenerator) DoCoverJSON(w io.Writer, params CoverHandlerParams) error {
-	return errors.New("DoCoverJSON is not implemented")
+	var err error
+	_, err = rg.prepareTemplateData(params)
+	if err != nil {
+		return fmt.Errorf("failed to prepare report data: %w", err)
+	}
+	c := &coverDatails{
+		Version: 1,
+	}
+	var jsonBytes []byte
+	if jsonBytes, err = json.Marshal(c); err != nil {
```

Comment:
Maybe it'd be easier to use https://pkg.go.dev/encoding/json@go1.19.13#Encoder?

Reply:
The documentation is very short there. Why is it interesting for us?
Does it prettyprint?

Reply:
> Why is it interesting for us?

It allows to dump JSON directly to the io.Writer - no need for intermediate buffers.

> Does it prettyprint?

It should be doable (if it's important -- I'm not sure we need it for out JSON interfaces).

Reply:
Got it. Thanks for sharing.

Reply:
It looks better. No branching, less lines.